pkg/cloudprovider/aws/object_storage_adapter.go

```diff
@@ -27,14 +27,16 @@ import (
 var _ cloudprovider.ObjectStorageAdapter = &objectStorageAdapter{}
 
 type objectStorageAdapter struct {
-	s3 *s3.S3
+	s3       *s3.S3
+	KMSKeyID string
```

Comment:
Very minor nit, but this should probably be package-local `kmsKeyID` rather than `KMSKeyID`.